onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -72,7 +81,31 @@ def _on_power_received(self, power):
         Args:
             power: ROS Twist message corresponding to desired powers in local reference frame
         """
-        self.power = utils.parse_twist(power)
+        parsed_power = utils.parse_twist(power)
+        if self.power_state_safety(parsed_power):
+            self.power = parsed_power
+
+    def power_state_safety(self, power):
+        # Compares power with controller limits
+        return_status = True
+        for axes in utils.get_axes():
+            if power[axes] > self.maxpower[axes]:
+                rospy.logerr(
+                    "===> Desired power exceeds maximum power in ", axes, "! Halting robot. <===")
+                self.pid_manager.soft_estop()
```

Comment:
Technically we need to set self.valid=False here. We can refactor self.valid into pid manager to avoid the need to do set self.valid, which I think is a better design. I'll make a commit in the fork that resolves this, so don't worry about it.